cacheops/simple.py

```diff
@@ -46,8 +48,20 @@ def wrapper(*args, **kwargs):
                 cache_key = 'c:' + _get_key(func, args, kwargs, extra)
                 try:
                     result = self.get(cache_key)
+
                 except CacheMiss:
-                    result = func(*args, **kwargs)
+                    if USE_LOCK:
+                        # This will block in all calls but the
+                        # first to avoid dog-pile effect.
+                        with redis_lock.Lock(self.conn, cache_key):
```

Comment:
OK. I've got the rest nearly finished, I will find the time to rebase before the end of week-end I hope. 

About the empty lines, I find the code more readable with, but no problem I will remove them. 

About the other mistakes (cache_set() outside of lock), my current workload & tiredness obviously affects the quality of the code I produce. Sorry for that, but above all, **thanks** for your careful review.